x-pack/test/functional/services/ml/data_frame_analytics_creation.ts

```diff
@@ -131,7 +131,9 @@ export function MachineLearningDataFrameAnalyticsCreationProvider(
       await this.assertSourceDataPreviewHistogramChartButtonCheckState(expectedDefaultButtonState);
       if (expectedDefaultButtonState === false) {
         await testSubjects.click('mlAnalyticsCreationDataGridHistogramButton');
-        await this.assertSourceDataPreviewHistogramChartButtonCheckState(true);
+        await this.assertSourceDataPreviewHistogramChartButtonCheckState(
+          expectedDefaultButtonState
+        );
```

Comment:
This change in `enableSourceDataPreviewHistogramCharts` looks wrong to me. The end result of this method should be an enabled/checked histogram preview button. We're validating the expected initial state upfront and if that's `false`, we're clicking the button to enable it, so I think it's correct to validate for `true` here and not for `expectedDefaultButtonState` (otherwise I think it would be `disabled -> click button -> expect disabled`). I think we have similar methods where the parameter is the desired state, but that's not the case here. So in it's current form, this method should only be used to enable the charts and not to toggle - we would need a separate method for this task.

Reply:
Updated [here](https://github.com/elastic/kibana/pull/125162/commits/7b6a8cdc9a228f5c5e1c2baca9c82ce7df4424c2)